files/usr/share/jeos-firstboot/raspberrywifi

```diff
@@ -0,0 +1,138 @@
+#!/bin/bash
+# Copyright (c) 2019 SUSE LLC
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+# Raspberry pi Wi-fi functions
+raspberrywifi_get_wlan_devices()
+{
+    list=()
+    local line
+    while read line; do
+        list+=("${line}" '')
+    done < <(ls -d /sys/class/net/*/wireless | awk -F'/' '{ print $5 }')
+    [ -n "$list" ]
+}
+
+raspberrywifi_get_wlan_networks()
+{
+    list=()
+    local line
+    while read line; do
+        list+=("${line}" '')
+    done < <(ip link set $wlan_device up && iwlist $wlan_device scan|grep ESSID|cut -d':' -f2|cut -d'"' -f2|sort -u)
+    [ -n "${list[*]}" ]
+}
+
+raspberrywifi_wlan_error()
+{
+    d --title $"Error" --msgbox "$1" 0 0
+    dialog --yesno $"Do you want to retry?" 0 0
+}
+
+# This is called by jeos-firstboot for user
+# interaction and access to the global systemd_firstboot_args array
+raspberrywifi_systemd_firstboot()
+{
+    if is_raspberry && ls -d /sys/class/net/*/wireless &>/dev/null; then
+        if dialog --yesno $"Configure wireless network?" 0 0; then
+            config_wireless=true
+        fi
+    fi
+    $config_wireless || return
```

Comment:
```suggestion
    [ "$config_wireless" = "true" ] || return
```

Reply:
Ah Github have too many options that I haven't used and this one is one of them. Corrected it by myself.

Reply:
I don't like it really, it doesn't even allow multline changes yet (or I haven't found a way to make those).